core/domain/user_services.py

```diff
@@ -680,36 +680,46 @@ def _save_user_settings(user_settings):
         user_settings: UserSettings domain object.
     """
     user_settings.validate()
-    user_models.UserSettingsModel(
-        id=user_settings.user_id,
-        gae_id=user_settings.gae_id,
-        email=user_settings.email,
-        role=user_settings.role,
-        username=user_settings.username,
-        normalized_username=user_settings.normalized_username,
-        last_agreed_to_terms=user_settings.last_agreed_to_terms,
-        last_started_state_editor_tutorial=(
+    user_model = user_models.UserSettingsModel.get_by_id(user_settings.user_id)
```

Comment:
Move this directly before the if on line 717.

Reply:
Done.